### DISCUSSION TITLE
JavaScript class names parser

### DISCUSSION THREAD

**Discussion by leonardofaria** [tailwindlabs/tailwindcss#6296] [Ideas]
Hi!

I am wondering if is there any project parsing and categorizing Tailwind CSS classes from a list. Ex:

```
font-sans → typography
font-serif → typography
font-medium → weight
font-thin → weight
pt-4 → padding, padding-top
py-4 → padding, padding-top, padding-bottom
```

It sounds a bit like reverse-engineering the config and themes, not sure if someone already did or if is there anything in the tailwindcss codebase that could be reusable for this need. 

Thanks

**Comment by olivermrose** [comment]
I don't see what the use case for this would be.

**Reply by leonardofaria** [reply]
By parsing these values it is possible to build a WYSIWYG HTML Editor, like [Fouita](https://editor-tw.fouita.com/)

**Comment by XEngine** [comment]
I know it's very old discussion, but it may help somebody still needs this solution. I've made a package exactly for this kind of use cases

https://github.com/XEngine/tailwindcss-class-parser
